clamav.py

```diff
@@ -205,6 +206,36 @@ def scan_file(path):
         signature = summary.get(path, AV_SIGNATURE_UNKNOWN)
         return AV_STATUS_INFECTED, signature
     else:
-        msg = "Unexpected exit code from clamscan: %s.\n" % av_proc.returncode
+        msg = "Unexpected exit code from clamdscan: %s.\n" % av_proc.returncode
         print(msg)
         raise Exception(msg)
+
+def start_clamd_daemon():
+    s3 = boto3.resource("s3")
+    s3_client = boto3.client("s3")
+
+    to_download = update_defs_from_s3(
+        s3_client, AV_DEFINITION_S3_BUCKET, AV_DEFINITION_S3_PREFIX
+    )
+
+    for download in to_download.values():
+        s3_path = download["s3_path"]
+        local_path = download["local_path"]
+        print("Downloading definition file %s from s3://%s" % (local_path, s3_path))
+        s3.Bucket(AV_DEFINITION_S3_BUCKET).download_file(s3_path, local_path)
+        print("Downloading definition file %s complete!" % (local_path))
+
+    av_env = os.environ.copy()
+    av_env["LD_LIBRARY_PATH"] = CLAMAVLIB_PATH
+
+    print("Starting clamd")
+
+    subprocess.Popen(
```

Comment:
From [the docs](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.wait)

> This will deadlock when using stdout=PIPE or stderr=PIPE and the child process generates enough output to a pipe such that it blocks waiting for the OS pipe buffer to accept more data. Use Popen.communicate() when using pipes to avoid that

I'm assuming this isn't a problem because the subprocess won't generate much data?

Also, I assume this is a fairly quick process, so we're not blocking the caller for long?

Reply:
correct. All the output goes into the log file. It takes 15 seconds on 2048 Lambda instance to load the virus databases